### PR TITLE
Move MetricData and add labels to the points

### PR DESCRIPTION
**Pull request by bogdandrutu**
The reason to move in `data` package is to avoid package circular dependencies, example if we move Aggregator to a `aggregators` package then:
`aggregators` depends on the `metrics` package for the `MetricData`.
`metrics` depends on the `aggregators` to use the Aggregator interface.

Split from #831

### CODE REVIEW

File: sdk/src/main/java/io/opentelemetry/sdk/metrics/data/MetricData.java [hunk 1]
```diff
@@ -43,51 +40,23 @@
...
+   * <p>Only one type of points are available at any moment for a {@link MetricData}, and the type
+   * is determined by the {@link Descriptor.Type}.
    *
    * @return the data {@link Point}s for this metric, or {@code null} if this type of points are not
    *     accepted.
    * @since 0.3.0
    */
-  @Nullable
-  public abstract Collection<LongPoint> getLongPoints();
+  public abstract Collection<Point> getPoints();
```
Comment (jkwatson) [inline sdk/src/main/java/io/opentelemetry/sdk/metrics/data/MetricData.java]: This is a good change. Glad to see it.

**Comment by bogdandrutu** [comment]
I would suggest that if we are ok with this change to do the same for SpanData (move to data package) to be consistent.

**Comment by jkwatson** [comment]
> I would suggest that if we are ok with this change to do the same for SpanData (move to data package) to be consistent.

Seems reasonable. Could you create an issue for it, so we don't lose track?

**Comment by bogdandrutu** [comment]
@jkwatson will do when I merge this (I want to give couple of hours to others to comment).

**Pull request merged**